### PR TITLE
<TreeView /> Improvements

### PR DESCRIPTION
**Pull request by liorzisman**
Includes:

* KeyCodes imported from `tree-view.tsx`

* TreeItem extracted to a separate file

* Removed tree- prefix from classnames

* Added a test that verifies that in the mobx version of the state management, modifying the tree data does not collapse the tree

### CODE REVIEW

File: src/components/auto-complete/auto-complete.tsx [hunk 1]
```diff
@@ -68,23 +68,23 @@ export class AutoComplete extends React.Component<AutoCompleteProps, AutoComplet
 
     private refCallback = (ref: HTMLInputElement) => {
         this.setState({input: ref});
-    };
+    }
```
Comment (liorzisman) [inline src/components/auto-complete/auto-complete.tsx]: automatic lint magic

File: src/components/tree-view/tree-node.st.css [hunk 2]
```diff
@@ -3,23 +3,23 @@
     -st-named: color_Keyboard_Focused, color_Active;
 }
 
-.tree-node {
+.node {
```
Comment (JoMarton) [inline src/components/tree-view/tree-node.st.css]: I don't get the subtlety of "node" vs. "item". Is there an actual difference? if not - I like `item`

File: src/components/tree-view/tree-item.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import nodeStyle from './tree-node.st.css';
```
[the remark this answers is not in the document]
Comment (liorzisman) [reply src/components/tree-view/tree-item.tsx]: renamed

File: src/components/tree-view/tree-node.st.css (hunk 2 above)
Comment (liorzisman) [reply src/components/tree-view/tree-node.st.css]: will make general clean up and improvements of naming across the component and its minions

File: src/components/tree-view/tree-item.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import {getFocusedItemKey, TreeItemProps, TreeItemData} from './tree-view';
```
[the remark this answers is not in the document]
Comment (liorzisman) [reply src/components/tree-view/tree-item.tsx]: `getFocusedItemKey` is a temporary function to be replaced, comment returned after mistakenly being deleted. I feel the interfaces belong with the TreeView as TreeItem is just the default implementation of a tree node using those interfaces. Will make a general rename of things to make it more clear across the entire component.

File: test/components/tree-view.spec.tsx [hunk 5]
```diff
@@ -189,6 +182,21 @@ describe('<TreeView />', () => {
...
+
+        const firstChildLabel = treeData[0].children![0].label;
+
+        expandItemWithLabel(select, treeData[0].label);
+
+        await waitForDom(() => expect(select(getTreeItem(firstChildLabel))).to.have.text(firstChildLabel));
+
+        (result as TreeViewMobxWrapper).renameLabel();
+
+        const changedLabel = 'Kaiserschmarrn';
```
[the remark this answers is not in the document]
Comment (liorzisman) [reply test/components/tree-view.spec.tsx]: i like that the assert reads that the tree item of `changedLabel` has the text `changedLabel` to reflect that it was changed, and not a literal value which does not fully reflect the intent

File: src/components/tree-view/tree-view.tsx [hunk 6]
```diff
@@ -60,66 +58,12 @@ export function initParentsMap(parentsMap: ParentsMap, data: TreeItemData[] = []
     });
 }
 
-const itemIdPrefix = 'TREE_ITEM';
-
 // This function is not perfect but for now this is
 // a solution that will be changed later
-function getFocusedItemKey(item: TreeItemData) {
+export function getFocusedItemKey(item: TreeItemData) {
```
Comment (JoMarton) [inline src/components/tree-view/tree-view.tsx]: remove this function

File: src/components/tree-view/tree-item.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
```
Comment (JoMarton) [inline src/components/tree-view/tree-item.tsx]: lets rename this file to `default-tree-item`

File: test/components/tree-view.spec.tsx [hunk 8]
```diff
@@ -1,31 +1,20 @@
-import * as keycode from 'keycode';
 import {observable} from 'mobx';
 import * as React from 'react';
 import {ClientRenderer, expect, simulate, sinon, waitFor} from 'test-drive-react';
 import {TreeViewDemo, TreeViewDemoCustom} from '../../demo/components/tree-view-demo';
-import {TreeItem, TreeView} from '../../src';
+import {TreeItem, TreeKeyCodes, TreeView} from '../../src';
 import {getLastAvailableItem, getNextItem, getPreviousItem} from '../../src/components/tree-view//tree-util';
```
Comment (JoMarton) [inline test/components/tree-view.spec.tsx]: double /?

File: test/components/tree-view.spec.tsx [hunk 9]
```diff
@@ -1,31 +1,20 @@
...
 import {initParentsMap, ParentsMap, TreeItemData, TreeStateMap} from '../../src/components/tree-view/tree-view';
```
Comment (JoMarton) [inline test/components/tree-view.spec.tsx]: add TreeView to the exported names

File: test/components/tree-view.spec.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (JoMarton) [reply test/components/tree-view.spec.tsx]: just have changedLabel as a member of `TreeViewMobxWrapper`

File: src/components/tree-view/tree-item.tsx (hunk 7 above)
Comment (liorzisman) [reply src/components/tree-view/tree-item.tsx]: done

File: src/components/tree-view/tree-view.tsx (hunk 6 above)
Comment (liorzisman) [reply src/components/tree-view/tree-view.tsx]: done

File: test/components/tree-view.spec.tsx (hunk 8 above)
Comment (liorzisman) [reply test/components/tree-view.spec.tsx]: cleared

File: test/components/tree-view.spec.tsx (hunk 9 above)
Comment (liorzisman) [reply test/components/tree-view.spec.tsx]: done

File: test/components/tree-view.spec.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (liorzisman) [reply test/components/tree-view.spec.tsx]: extracted changedLabel as a const in the file since its used also in another place

**Pull request merged**